python/mxnet/text/embedding.py

```diff
@@ -0,0 +1,600 @@
+class FastText(TokenEmbedding):
+    """The fastText token embedding.
+
+    FastText is an open-source, free, lightweight library that allows users to
+    learn text representations and text classifiers. It works on standard,
+    generic hardware. Models can later be reduced in size to even fit on mobile
+    devices. (Source from https://fasttext.cc/)
+
+    References:
+    Enriching Word Vectors with Subword Information
+    Piotr Bojanowski, Edouard Grave, Armand Joulin, and Tomas Mikolov
+    https://arxiv.org/abs/1607.04606
+
+    Bag of Tricks for Efficient Text Classification
+    Armand Joulin, Edouard Grave, Piotr Bojanowski, and Tomas Mikolov
+    https://arxiv.org/abs/1607.01759
+
+    FastText.zip: Compressing text classification models
+    Armand Joulin, Edouard Grave, Piotr Bojanowski, Matthijs Douze, Herve Jegou,
+    and Tomas Mikolov
+    https://arxiv.org/abs/1612.03651
+
+    Website:
+    https://fasttext.cc/
+
+    To get the updated URLs to the externally hosted pre-trained token embedding
+    files, visit
+    https://github.com/facebookresearch/fastText/blob/master/
+    pretrained-vectors.md
+
+
+    Parameters
+    ----------
+    pretrain_file : str, default 'wiki.en.vec'
+        The name of the pre-trained token embedding file.
+    embed_root : str, default '~/.mxnet/embeddings/'
+        The root directory for storing embedding-related files.
+    unknown_vec : callback
+        The callback used to initialize the embedding vector for the unknown
+        token.
+    """
+
+    # Map a pre-trained token embedding file and its SHA-1 hash.
+    pretrained_file_name_sha1 = \
+        {'wiki.en.vec': 'c1e418f144ceb332b4328d27addf508731fa87df',
+         'wiki.simple.vec': '55267c50fbdf4e4ae0fbbda5c73830a379d68795',
+         'wiki.zh.vec': '117ab34faa80e381641fbabf3a24bc8cfba44050'}
+    url_prefix = 'https://s3-us-west-1.amazonaws.com/fasttext-vectors/'
+
+    def __init__(self, pretrained_file_name='wiki.en.vec',
+                 embedding_root='~/.mxnet/embeddings/',
+                 init_unknown_vec=nd.zeros, **kwargs):
+        FastText._check_pretrained_file_names(pretrained_file_name)
+        url = FastText.url_prefix + pretrained_file_name
+
+        super(FastText, self).__init__(**kwargs)
+
+        pretrained_file_path = FastText._get_pretrained_file_path_from_url(
+            url, embedding_root, pretrained_file_name)
+
+        self._load_embedding(pretrained_file_path, ' ', init_unknown_vec)
+
+
+class CustomEmbedding(TokenEmbedding):
+    """User-defined token embedding.
+
+    This is to load embedding vectors from a user-defined pre-trained text
+    embedding file.
+
+    Denote by v_ij the j-th element of the token embedding vector for token_i,
+    the expected format of a custom pre-trained token embedding file is:
+
+    token_1`elem_delim`v_11`elem_delim`v_12`elem_delim`...`elem_delim`v_1k\n
+    token_2`elem_delim`v_21`elem_delim`v_22`elem_delim`...`elem_delim`v_2k\n
+    ...
```

Comment:
Fix formatting. Currently it looks like this http://mxnet-doc.s3-accelerate.dualstack.amazonaws.com/api/python/text/text.html#mxnet.text.embedding.CustomEmbedding

You can probably make the format into code blocks. Also I don't think back-ticks are part of the format, so you may want to explain that in the comment.

Reply:
Resolved.